Mathlib/Algebra/Order/Monoid/Defs.lean

```diff
@@ -137,30 +137,56 @@ class LinearOrderedCancelCommMonoid (α : Type*) extends OrderedCancelCommMonoid
 
 attribute [to_additive existing] LinearOrderedCancelCommMonoid.toLinearOrderedCommMonoid
 
+/--
+The additive equivalent of `MulZeroClass` for `Top`
```

Comment:
Where should it go, then?

Reply:
Ah, `LinearOrderedAddCommMonoidWithTop` is misplaced too. If you open a PR moving it to `Algebra.Order.MonoidWithTop`, I will approve immediately.

Reply:
Should `Mathlib/Algebra/Order/Monoid/WithTop.lean` be moved to `Mathlib/Algebra/Order/MonoidWithTop/Basic.lean`, or just the `LinearOrderedAddCommMonoidWithTop` instance?

Reply:
Perhaps all of `GroupWithZero` could just be additivized?

Reply:
I just noticed that `LinearOrderedAddCommGroupWithTop` is in `Algebra.Order.Group` while the rest of the stuff there is tagged `to_additive`, so presumably it should be moved as well.

Reply:
> Should `Mathlib/Algebra/Order/Monoid/WithTop.lean` be moved to `Mathlib/Algebra/Order/MonoidWithTop/Basic.lean`, or just the `LinearOrderedAddCommMonoidWithTop` instance?

I've just looked at it and some parts are additivised and some not. The "some not" should move. The rest can stay.

Reply:
> Perhaps all of `GroupWithZero` could just be additivized? How hard would it be to extend `to_additive` and make it able to create two new versions in these cases, one for `Top` and one for `Bot`?

That's completely irrelevant. Even if we could make `to_additive` do that, this wouldn't change the fact that it's not a theory that has additive and multiplicative counterparts

Reply:
Isn't it? I'd say that `instLinearOrderedAddCommMonoidWithTopAdditiveOrderDual` and `instLinearOrderedAddCommGroupWithTopAdditiveOrderDual` show that it has additive and multiplicative counterparts (at least up to an order dual, which can be avoided by using `Bot` instead of `Top`).

Reply:
Still, that's a very different "additivisation" than what you see in the rest of the library, and it shouldn't be considered along with it. Nowhere else do we additivise `0` to `\top`.

Reply:
I understand. However, just copying `Mathlib.Algebra.Order.GroupWithZero.Canonical` and manually editing everything to use `Top` and `Bot` doesn't sit well with me

Reply:
You can try writing the metaprogram that does this for you, but it's probably strictly harder than doing it yourself and also won't be that applicable as I have mentioned

Reply:
I created #14667, although I'm really not sure about it.